### PR TITLE
[Feature Store] Block dataframe as source when running remotely

### PR DESCRIPTION
**Pull request by katyakats**
https://jira.iguazeng.com/browse/ML-1346

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -386,6 +394,10 @@ def ingest(
             "featureset.spec.engine must be set to 'spark' to ingest with spark"
         )
     if featureset.spec.engine == "spark":
+        if isinstance(source, pd.DataFrame):
```
Comment (urihoenig) [inline mlrun/feature_store/api.py:397]: Why? What about simple ingest with spark?
https://docs.mlrun.org/en/latest/feature-store/transformations.html#using-spark-execution-engine

**Review by urihoenig — CHANGES_REQUESTED** [review]

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (katyakats) [reply mlrun/feature_store/api.py:397]: with a df as a source it would fail

File: mlrun/feature_store/api.py [hunk 2]
```diff
@@ -393,7 +393,7 @@ def ingest(
         raise mlrun.errors.MLRunInvalidArgumentError(
             "featureset.spec.engine must be set to 'spark' to ingest with spark"
         )
-    if featureset.spec.engine == "spark":
+    if featureset.spec.engine == "spark" and run_config is not None:
```
Comment (urihoenig) [inline mlrun/feature_store/api.py:396]: But now it'll just continue to a non-spark ingest.
It has to be added in the if isinstance  in the next line.

**Review by urihoenig — CHANGES_REQUESTED** [review]

**Review by urihoenig — APPROVED** [review]

File: mlrun/feature_store/api.py [hunk 3]
```diff
@@ -311,6 +315,10 @@ def ingest(
 
     if mlrun_context:
         # extract ingestion parameters from mlrun context
+        if isinstance(source, pd.DataFrame):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "DataFrame source is illegal when running ingest with mlrun context"
```
Comment (Hedingber) [inline mlrun/feature_store/api.py:320]: "with mlrun context" won't mean anything to a user, explain it in their words

File: tests/system/feature_store/test_feature_store.py [hunk 4]
```diff
@@ -82,6 +82,18 @@ def _ingest_stocks_featureset(self):
...
+        try:
+            df = fs.ingest(
+                stocks_set,
+                stocks,
+                infer_options=fs.InferOptions.default(),
+                run_config=fs.RunConfig(local=True),
+            )
+            assert False
+        except mlrun.errors.MLRunInvalidArgumentError:
+            pass
```
Comment (Hedingber) [inline tests/system/feature_store/test_feature_store.py:95]: The way to assert something is failed is `with pytest.raises(...` we use it a lot, search the code base

File: tests/system/feature_store/test_spark_engine.py [hunk 5]
```diff
@@ -87,6 +88,30 @@ def test_basic_remote_spark_ingest(self):
...
+                "name": ["Jean", "Jacques", "Pierre"],
+                "last_name": ["Dubois", "Dupont", "Lavigne"],
+            }
+        )
+
+        measurements = fs.FeatureSet(
+            "measurements", entities=[fs.Entity("name")], engine="spark",
+        )
+
+        try:
```
Comment (Hedingber) [inline tests/system/feature_store/test_spark_engine.py:103]: same

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/feature_store/api.py (hunk 3 above)
Comment (katyakats) [reply mlrun/feature_store/api.py:320]: mlrun context is a parameter that the user passes to ingest function

**Pull request merged**